### ISSUE TITLE
iex overrides stdin to /dev/tty 

### ISSUE THREAD

**Issue by pragdave** [elixir-lang/elixir#8734]
I _think_ I know why this is happening, and I see why it might be hard to fix given the layers involved, but this isn't well-behaved.

Pipe a command into `iex`, and it reads from the pipe.

~~~ 
$ echo h h | iex

Interactive Elixir (1.8.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> 
                                    def h()                                     

Prints the documentation for IEx.Helpers.

etc...
~~~

However, at EOF on the pipe, it switches to reading from `/dev/tty`, and `iex(2)>` appears on the console.

At this point, a single ^C will terminate iex, all the way back to the shell prompt.

~~~
iex(2)> ^C
BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution

$ 
~~~

I'm guessing the underlying Erlang User stopped when it got the eof, but iex kept running?

Cheers

Dave

**Comment by josevalim** [comment]
Good catch @pragdave! Recently fixed on #8560. Thanks!

**Comment by pragdave** [comment]
Oops: I didn't see that one. Sorry for the duplicate
